dask/dataframe/groupby.py

```diff
@@ -1284,7 +1285,19 @@ def _cum_agg(self, token, chunk, aggregate, initial):
         """Wrapper for cumulative groupby operation"""
         meta = chunk(self._meta)
         columns = meta.name if is_series_like(meta) else meta.columns
-        by = self.by if isinstance(self.by, list) else [self.by]
+        by_cols = self.by if isinstance(self.by, list) else [self.by]
+
+        # rename "by" columns internally
+        # to fix cumulative operations on the same "by" columns
+        # ref: https://github.com/dask/dask/issues/9313
+        if columns is not None and set(columns).intersection(set(by_cols)):
+            by = []
+            for col in by_cols:
+                suffix = str(uuid.uuid4())
+                self.obj = self.obj.assign(**{col + suffix: self.obj[col]})
```

Comment:
We need to update both `self.obj` as well as the list of `by` columns -- that's why I didn't do:

```python
self.obj = self.obj.assign(
    **{
        col + str(uuid.uuid4()): self.obj[col]
        for col in by_cols
    }
)
```